tests/fairship_tests/harness.py

```
@@ -0,0 +1,306 @@
+        elif current_selector in {"*", test_name}:
+            patterns.append(line)
+
+    return [_compile_wildcard(pattern) for pattern in patterns]
+
+
+def run(test_name: str, workdir: Path) -> tuple[int, str]:
+    """Run one configured test, returning its status and normalized output."""
+    case = _test_case(test_name)
+    environment = os.environ.copy()
+    environment["FAIRSHIP_ROOT"] = str(REPOSITORY_ROOT)
+    result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)